wolfcrypt/src/port/silabs/silabs_ecc.c

```diff
@@ -0,0 +1,319 @@
+    key->key.size = key->dp->size;
+    key->key.storage.method = SL_SE_KEY_STORAGE_EXTERNAL_PLAINTEXT;
+    key->key.flags = SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PRIVATE_KEY
+        | SL_SE_KEY_FLAG_ASYMMMETRIC_SIGNING_ONLY
+        ;
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    key->key.storage.location.buffer.pointer = key->key_raw;
+    if (sl_stat != SL_STATUS_OK)
+          return WC_HW_E;
+
+    return wc_export_int(&key->k, key->key.storage.location.buffer.pointer,
+                         &keySz, keySz,
+                         WC_TYPE_UNSIGNED_BIN);
+}
+
+int silabs_ecc_sig_to_rs(ecc_key* key, word32 keySz)
+{
+    sl_status_t sl_stat;
+    int err = MP_OKAY;
+
+    key->key.type = silabs_map_key_type(key->dp->id);
+    key->key.size = keySz;
+    key->key.storage.method = SL_SE_KEY_STORAGE_EXTERNAL_PLAINTEXT;
+    key->key.flags = SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PUBLIC_KEY
+        | SL_SE_KEY_FLAG_ASYMMMETRIC_SIGNING_ONLY
+        ;
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    key->key.storage.location.buffer.pointer = key->key_raw;
+    if (sl_stat != SL_STATUS_OK)
+          return WC_HW_E;
+
+    if (err == MP_OKAY) {
+        keySz = key->dp->size;
+        err = wc_export_int(key->pubkey.x,
+                            key->key.storage.location.buffer.pointer,
+                            &keySz, keySz, WC_TYPE_UNSIGNED_BIN);
+        if (err == MP_OKAY)
+            err = wc_export_int(key->pubkey.y,
+                                key->key.storage.location.buffer.pointer + keySz,
+                                &keySz, keySz, WC_TYPE_UNSIGNED_BIN);
+    }
+
+    return err;
+}
+
+int silabs_ecc_import_private_raw(ecc_key* key, word32 keySz, const char* d, int encType)
+{
+    sl_status_t sl_stat;
+    int err = MP_OKAY;
+    key->type = ECC_PRIVATEKEY;
+    key->key.flags |= SL_SE_KEY_FLAG_ASYMMETRIC_BUFFER_HAS_PRIVATE_KEY;
+
+    sl_stat = sl_se_get_storage_size(&key->key, &key->key.storage.location.buffer.size);
+    if (sl_stat != SL_STATUS_OK)
+          return WC_HW_E;
+
+    if (encType == WC_TYPE_HEX_STR)
+        err = mp_read_radix(&key->k, d, MP_RADIX_HEX);
+    else
+        err = mp_read_unsigned_bin(&key->k, (const byte*)d,
+                                   key->dp->size);
+    if (err == MP_OKAY) {
+        err = wc_export_int(&key->k, key->key.storage.location.buffer.pointer + (2 * keySz),
+                            &keySz, keySz,
+                            WC_TYPE_UNSIGNED_BIN);
+    }
+
+    return err;
+}
+
+int silabs_ecc_shared_secret(ecc_key* private_key, ecc_key* public_key,
+                             byte* out, word32* outlen)
+{
+    sl_se_command_context_t cmd;
+    sl_se_key_descriptor_t key_out;
+    sl_se_key_descriptor_t pub_key;
+
+    uint32_t pub_sz = 0;
+    sl_status_t sl_stat;
+
+    pub_key = public_key->key;
+    pub_key.flags = 0x2000;
```

Comment:
What is this `flags` `0x2000`? Is there an enum/macro we can use? If not please add and comment.